### PR TITLE
Update and refactor of all roles and permissions functions

### PR DESCRIPTION
**Pull request by Sampurna5**

### CODE REVIEW

File: pkg/permission.go [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+*/
+func DeletePermission(db *gorm.DB, permissionID uint) error {
+	transaction := db.Begin()
+	// Check if the permission with given permission name exists
+	permission, err := GetPermissionByID(db, permissionID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return utils.ErrPermissionNotFound
+		}
+
```
Comment (leodahal4) [inline pkg/permission.go:140]: return err

File: pkg/permission.go [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+	}
+
+	// Check if the permission is assigned to a role. If yes, delete the association object
+	if err = transaction.Where("permission_id = ?", permission.ID).First(&models.RolePermission{}).Error; err == nil {
+		if err = transaction.Where("permission_id = ?", permission.ID).Delete(&models.UserPermissionInt{}).Error; err != nil {
```
Comment (leodahal4) [inline pkg/permission.go:145]: remove nested codes

File: pkg/permission.go [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+			transaction.Rollback()
+			return err
+		}
+	}
+
+	// Delete the permission
+	db.Where("id = ?", permissionID).Delete(&permission)
```
Comment (leodahal4) [inline pkg/permission.go:152]: return 	db.Where("id = ?", permissionID).Delete(&permission).Error

File: pkg/permission.go [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+*/
+func AssignRolePermissions(db *gorm.DB, roleID uint, permissionIDs []uint) error {
+	// Check if the role with given roleName exists
+	var role models.Role
+
+	err := db.Where("id = ?", roleID).First(&role).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return utils.ErrRoleNotFound
+		}
```
Comment (leodahal4) [inline pkg/permission.go:173]: return err

File: cmd/roles_and_permissions/authority.go [hunk 5]
```diff
@@ -197,7 +198,7 @@ AssignRolePermissions assign a list of permissions to a specified role.
 It takes in two parameters: roleID of type uint and permissionIDs of type slice of uint and return
 an error object.
 */
-func (a *Authority) AssignRolePermissions(roleID uint, permissionIDs []uint) error {
+func (a *Authority) AssignRolePermissions(roleID uint, permissionIDs []uint) []string {
```
Comment (leodahal4) [inline cmd/roles_and_permissions/authority.go:201]: its better to change the Authority repo -> authority, as lower case.

**Review by leodahal4 — APPROVED** [review]

File: cmd/roles_and_permissions/authority.go (hunk 5 above)
Comment (Sampurna5) [reply cmd/roles_and_permissions/authority.go:201]: Sure, thanks.

**Pull request merged**